packages/vision/src/helpers.js

```diff
@@ -73,14 +73,27 @@ var coerceImage = (image, callback) => {
  * @return {Function} - The function that, when called, will call annotateImage
  *   asking for the single feature annotation.
  */
-var _createSingleFeatureMethod = featureValue => {
-  return function(image, options) {
-    return this.annotateImage({
-      image: image,
-      features: [{type: featureValue}],
-    }, options);
-  };
-};
+ var _createSingleFeatureMethod = featureValue => {
+   return function(annotateImageRequest, callOptions) {
+     annotateImageRequest.features = annotateImageRequest.features || [];
+
+     // Ensure the feature value indicated by the user's method choice exists on the
+     // features array; if it does not, add it.
+     var found = false;
```

Comment:
Okay, so let's talk through the situation where someone calls:

```es6
vision.cropHints({
  image: {...},
  features: [{type: features.FACE_DETECTION}],
});
```

There are four options:

  1. Take the union of the feature designated by the method and the features designated in the `features` array. (This is what I implemented.)
  2. Take the feature designated by the method only, throw away the specified `features` array. (This is what you suggest.)
  3. Take the feature designated by the `features` array, throw away the method designation. (This is what @jmdobry's sample code in #2553 did.)
  4. Throw an exception.

I believe that choices (2) and (3) both violate the principle of least surprise: as a user, I wonder why I did not get something that I asked for. One might argue that nobody would ever do this intentionally, but I believe there might be a case where extant code calling `cropHints` exists, and someone decides they _also_ want faces, and in looking up how to do it in docs, stumbles on the `features` array. It is not hard to imagine this scenario.

Therefore, I assert that the only acceptable options are (1) and (4). If we do (4), basically we throw an exception and direct users to `annotateImage`. The advantage to this is that it is explicit: users will not ever be in a situation where they are charged for N features when they only wanted N - 1. On the other hand, we create more cognitive work for the developer.

(1) seems like the best option to me. It passes principle of least surprise as well as holding to Postel's Law.